rules/no-import-test-files.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+const path = require('path');
+const arrify = require('arrify');
+const pkgUp = require('pkg-up');
+const multimatch = require('multimatch');
+const util = require('../util');
+
+const defaultFiles = [
+	'test.js',
+	'test-*.js',
+	'test/**/*.js',
+	'**/__tests__/**/*.js',
+	'**/*.test.js'
+];
+
+function isTestFile(files, rootDir, sourceFile, importedFile) {
+	const absoluteImportedPath = path.resolve(path.dirname(sourceFile), importedFile);
+	const relativePath = path.relative(rootDir, absoluteImportedPath);
+
+	return multimatch([relativePath], files).length === 1;
+}
+
+function getProjectInfo() {
+	const packageFilePath = pkgUp.sync();
+
+	return {
+		rootDir: packageFilePath && path.dirname(packageFilePath),
+		files: util.getAvaConfig(packageFilePath).files
+	};
+}
+
+const create = context => {
+	const filename = context.getFilename();
+
+	if (filename === '<text>') {
+		return {};
+	}
+
+	const projectInfo = getProjectInfo();
+	const options = context.options[0] || {};
+	const files = arrify(options.files || projectInfo.files || defaultFiles);
+
+	if (!projectInfo.rootDir) {
+		// Could not find a package.json folder
+		return {};
+	}
+
+	return {
+		ImportDeclaration: node => {
+			const isImportingTestFile = isTestFile(files, projectInfo.rootDir, filename, node.source.value);
+
+			if (isImportingTestFile) {
+				context.report({
+					node,
+					message: `You are importing a test file`,
+				});
+			}
```

Comment:
Line 50 to 57 is almost duplicated with the `CallExpression` one. Just make an abstracted method that accepts the `value`.